src/porting.cpp

```diff
@@ -738,10 +738,12 @@ void attachOrCreateConsole()
 {
 #ifdef _WIN32
 	static bool consoleAllocated = false;
-	const bool redirected = (_fileno(stdout) == -2 || _fileno(stdout) == -1); // If output is redirected to e.g a file
-	if (!consoleAllocated && redirected && (AttachConsole(ATTACH_PARENT_PROCESS) || AllocConsole())) {
-		freopen("CONOUT$", "w", stdout);
-		freopen("CONOUT$", "w", stderr);
+	const bool redirected = _fileno(stdout) >= 0; // If output is redirected to e.g a file
+	if (!consoleAllocated && !redirected) {
+		if (AttachConsole(ATTACH_PARENT_PROCESS) || AllocConsole()) {
+			freopen("CONOUT$", "w", stdout);
+			freopen("CONOUT$", "w", stderr);
+		}
 		consoleAllocated = true;
```

Comment:
At first it seemed like the variable should be in the code block above. Turns out it's a slightly misleading name.
Might you please be so nice to change it to `redirect_attempted`, `run_once`, or similar?